trieste/acquisition/function.py

```diff
@@ -676,3 +676,245 @@ def batch_ei(at: TensorType) -> TensorType:
             return tf.reduce_mean(batch_improvement, axis=-1, keepdims=True)  # [..., 1]
 
         return batch_ei
+
+
+class GreedyAcquisitionFunctionBuilder(ABC):
+    """
+    An :class:`GreedyAcquisitionFunctionBuilder` builds an acquisition function
+    suitiable for greedily building batches for batch Bayesian
+    Optimisation. :class:`GreedyAcquisitionFunctionBuilder` differs
+    from :class:`AcquisitionFunctionBuilder` by requiring that a set
+    of pending points is passed to the builder.
+    """
+
+    @abstractmethod
+    def prepare_acquisition_function(
+        self,
+        datasets: Mapping[str, Dataset],
+        models: Mapping[str, ProbabilisticModel],
+        pending_points: TensorType = None,
```

Comment:
Strictly speaking, pending points should be marked as optional:

```suggestion
        pending_points: Optional[TensorType] = None,
```